integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py

```diff
@@ -1,24 +1,30 @@
 import json
 import logging
-import os
 from typing import Any, ClassVar, Dict, List, Optional
 
+import boto3
 import requests
+from botocore.exceptions import BotoCoreError
 from haystack import component, default_from_dict, default_to_dict
-from haystack.lazy_imports import LazyImport
+from haystack.utils import Secret, deserialize_secrets_inplace
+
 from haystack_integrations.components.generators.amazon_sagemaker.errors import (
     AWSConfigurationError,
     SagemakerInferenceError,
     SagemakerNotReadyError,
 )
 
-with LazyImport(message="Run 'pip install boto3'") as boto3_import:
-    import boto3  # type: ignore
-    from botocore.client import BaseClient  # type: ignore
-
 
 logger = logging.getLogger(__name__)
 
+AWS_CONFIGURATION_KEYS = [
+    "aws_access_key_id",
+    "aws_secret_access_key",
+    "aws_session_token",
+    "aws_region_name",
+    "aws_profile_name",
+]
```

Comment:
Would be better if we could do without this constant, if we want to print them in the error message in `get_aws_session` we should reference the arguments explicitly.